packages/lisk-p2p/src/p2p_types.ts

```diff
@@ -66,9 +67,16 @@ export interface P2PNodeInfo {
 	readonly options?: P2PInfoOptions;
 }
 
+export interface P2PClosePacket {
+	readonly peerInfo: P2PPeerInfo,
```

Comment:
I think at this point, peer which is getting closed will have `P2PDiscoveredPeerInfo` so maybe we can use that?

Reply:
I also thought this before but it is not guaranteed because we don't get the peer's full `P2PDiscoveredPeerInfo` until after we successfully made a `peer.fetchStatus` call to them. If the socket disconnects before we get the response from the `peer.fetchStatus` call, then we won't have the full `P2PDiscoveredPeerInfo`.